eth/backend.go

```diff
@@ -857,9 +857,19 @@ func (s *Ethereum) Init(stack *node.Node, config *ethconfig.Config) error {
 
 	s.apiList = jsonrpc.APIList(chainKv, ethRpcClient, txPoolRpcClient, miningRpcClient, ff, stateCache, blockReader, s.agg, httpRpcCfg, s.engine, s.logger)
 	go func() {
-		if err := cli.StartRpcServer(ctx, httpRpcCfg, s.apiList, s.logger); err != nil {
-			s.logger.Error(err.Error())
-			return
+		if config.SilkwormEnabled && httpRpcCfg.Enabled {
+			err = s.silkworm.StartRpcDaemon(chainKv)
+			if err != nil {
+				s.logger.Error(err.Error())
+				return
+			}
+			defer s.silkworm.StopRpcDaemon()
```

Comment:
maybe you did mean: 
```
select {
case <-ctx.Done(): 
   s.silkworm.StopRpcDaemon()
}
```
?

Reply:
I did it that way because the existing `StartRpcServer` seems to block the calling goroutine, so I replicated the same behaviour; in other words, `StartRpcServer` calls `startRegularRpcServer`, which uses a pattern similar to mine (i.e. defer + `<- ctx.Done()`).

But maybe it's an anti-pattern, so I will change as you suggest.

Reply:
Yes, and if it blocks - then only way to get to this line of code - “exit without error” - which means  server already don’t need shutdown. 

In my head “StopRpcDaemon” can gracefuly stop working server. We need call  it at “Ctrl+C”

Reply:
No, because `s.silkworm.StartRpcDaemon` does _**not**_ block (differently from `cli.StartRpcServer`). But it works in both ways, so I changed the code

Reply:
ah, ok. can add it to name like "StartInBackground" - because most of go's funcs are blocking.